test/test.js

```diff
@@ -1059,10 +1059,17 @@ describe("S3rver Tests", function() {
     expect(find(data.Deleted, { Key: "key67" })).to.exist;
   });
 
-  it("should not throw when using deleteObjects with zero objects", function*() {
-    yield s3Client
-      .deleteObjects({ Bucket: buckets[2], Delete: { Objects: [] } })
-      .promise();
+  it("should report invalid XML when using deleteObjects with zero objects", function*() {
+    let error;
+    try {
+      yield s3Client
+        .deleteObjects({ Bucket: buckets[2], Delete: { Objects: [] } })
+        .promise();
+    } catch (err) {
+      error = err;
+    }
+    expect(error).to.exist;
+    expect(error.code).to.equal("MalformedXML");
```

Comment:
@kherock Do you think this could break users' code? I can imagine a naive `deleteObject` user might make empty requests sometimes. Should this be released in a patch, minor, or major release?

Reply:
As of a few patches ago, I'm pretty certain the server would just throw an unhandled error inside that callback function, so bringing this up to spec shouldn't be a big deal. I vote patch release.